### DISCUSSION TITLE
Generating protobufs

### DISCUSSION THREAD

**Discussion by lrq619** [vhive-serverless/vSwarm-proto#57] [Q&A] [answered]
system image: ubuntu20.04
go version: go1.20.5 linux/amd64
Bug description: make proto-all-go failed with error
Reproduce steps:
```
git clone git@github.com:vhive-serverless/vSwarm-proto.git
cd vSwarm-proto/
make proto-all-go
```
Error message:
```
protoc --go_out=. \
        --go_opt=paths=source_relative \
        --go-grpc_out=. \
        --go-grpc_opt=paths=source_relative \
        ./proto/*/*.proto
2023/06/14 07:43:35 protoc-gen-go: error:inconsistent package import paths: "github.com/vhive-serverless/vSwarm-proto/proto/aes", "github.com/vhive-serverless/vSwarm-proto/proto/auth"
--go_out: protoc-gen-go: Plugin failed with status code 1.
make: *** [Makefile:14: proto-all-go] Error 1
```
I then turned to simply run protoc for one specific benchmark but it also failed, for example:
```
protoc --go_out=. \
        --go_opt=paths=source_relative \
        --go-grpc_out=. \
        --go-grpc_opt=paths=source_relative \
        ./proto/aes/aes.proto
```
and the error message shows:
```
protoc-gen-go-grpc: program not found or is not executable
--go-grpc_out: protoc-gen-go-grpc: Plugin failed with status code 1.
```

I have tried:
```
go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.26
go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.1
```
But it doesn't solve the problem

**Comment by ustiugov** [comment]
@dhschall can you take a look?

one possible problem is that you use Golang 1.20 while vSwarm relies on 1.18 as I can see from go.mod files there.

**Comment by dhschall** [comment, marked as answer]
The problem is you also need to install the protoc compiler. The two packes you installed are only plugins.
https://grpc.io/docs/languages/go/quickstart/

You can use the workflow file as a hint.
https://github.com/vhive-serverless/vSwarm-proto/blob/d001f860c40d52ef38804abf6dd5321dfc37f767/.github/workflows/build-tests.yml#L43
```
PROTOC_ZIP=protoc-3.14.0-linux-x86_64
curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v3.14.0/$PROTOC_ZIP.zip
unzip -o $PROTOC_ZIP.zip -d $HOME/.local
chmod +x $HOME/.local/bin/protoc
rm -f $PROTOC_ZIP.zip
```

**Comment by ustiugov** [comment]
@lrq619 could you please try and document the procedure?

**Reply by lrq619** [reply]
Yes, I have wrote the procedure in this file: https://github.com/lrq619/vSwarm/blob/add_bert_benchmark/docs/adding_standalone_benchmark.md

**Comment by dhschall** [comment]
I guess this is solved?
Reopen if not
